File: BidService/Domain/Entities/Bid.cs
using Common.Domain.Entities;
using Common.Domain.Enums;

namespace BidService.Domain.Entities
{
    public class Bid : BaseEntity
    {
        public Guid AuctionId { get; set; }
        public Guid BidderId { get; set; }
        public string BidderUsername { get; set; } = string.Empty;
        public decimal Amount { get; set; }
        public DateTimeOffset BidTime { get; set; }
        public BidStatus Status { get; set; }
    }

    public enum BidStatus
    {
        Pending,
        Accepted,
        AcceptedBelowReserve,
        TooLow,
        Rejected
    }
}

File: Web/types/bid.ts
export interface Bid {
  id: string;
  auctionId: string;
  bidderId: string;
  bidderUsername: string;
  amount: number;
  bidTime: string;
  status: BidStatus;
  errorMessage?: string;
  minimumNextBid: number;
  minimumIncrement: number;
  createdAt: string;
  updatedAt?: string;
}

export enum BidStatus {
  Pending = "Pending",
  Accepted = "Accepted",
  AcceptedBelowReserve = "AcceptedBelowReserve",
  TooLow = "TooLow",
  Rejected = "Rejected"
}

export interface PlaceBidDto {
  auctionId: string;
  amount: number;
}
